### PR TITLE
[gpu] Make collective permute decomposer accept no channel ids

### PR DESCRIPTION
**Pull request by shraiysh**
When no channel ids are present, this patch creates sends and recvs with channel id 0.

### CODE REVIEW

**Comment by shraiysh** [comment]
@bixia1 It wasn't clear to me why we decompose collective permute when we have no opportunity of pipelining it [here](https://github.com/shraiysh/xla/blob/main/xla/service/collective_permute_decomposer_test.cc#L73-L129)?

**Comment by shraiysh** [comment]
The failures are unrelated - HEAD is failing with these errors.

File: xla/service/collective_permute_decomposer_test.cc [hunk 1]
```diff
@@ -282,38 +276,50 @@ TEST_F(CollectivePermuteDecomposerTest, ForwardPipeline2) {
...
+                                       {"[[CHANNEL_ID_2]]", "channel_id=2, "}});
+  std::string no_channel_id_hlo = absl::StrReplaceAll(
+      kModuleStr, {{"[[CHANNEL_ID_1]]", ""}, {"[[CHANNEL_ID_2]]", ""}});
+
+  auto check_module = [](const std::string module_str,
+                         absl::flat_hash_set<int64_t> channel_ids) {
+    TF_ASSERT_OK_AND_ASSIGN(std::unique_ptr<HloModule> module,
+                            ParseAndReturnUnverifiedModule((module_str)));
+    CollectivePermuteDecomposer decomposer(/*threshold_in_bytes=*/0);
+    TF_ASSERT_OK_AND_ASSIGN(bool changed, decomposer.Run(module.get()));
```
Comment (frgossen) [inline xla/service/collective_permute_decomposer_test.cc:290]: Did you consider a filecheck test here? Will leave it up to you if you want to change this.

File: xla/service/collective_permute_decomposer_test.cc [hunk 2]
```diff
@@ -369,60 +375,72 @@ TEST_F(CollectivePermuteDecomposerTest, ForwardPipelineWithMatmul) {
...
+    EXPECT_TRUE(changed);
+    HloModule* transformed_module = module.get();
+    // Check the annotations and ordering of the decomposed send-recv pairs.
+    // We expect the recv to come before the send in the while body, both for
+    // the forward edge ({0,1},{1,2},{2,3}}) and the backward edge ({3,0}). This
+    // is an XLA invariant that shouldn't be broken (see
+    // https://openxla.org/xla/operation_semantics#send for details of the
+    // semantics).
```
Comment (frgossen) [inline xla/service/collective_permute_decomposer_test.cc:397]: same

**Review by frgossen — COMMENTED** [review]
Thank you!!

**Review by frgossen — APPROVED** [review]

File: xla/service/collective_permute_decomposer.cc [hunk 3]
```diff
@@ -142,13 +137,15 @@ absl::Status DecomposeCollectivePermute(
 
   HloInstruction* after_all =
       computation->AddInstruction(HloInstruction::CreateToken());
-  HloInstruction* recv = computation->AddInstruction(
-      HloInstruction::CreateRecv(data_shape, after_all, channel_id));
+  HloInstruction* recv = computation->AddInstruction(HloInstruction::CreateRecv(
+      /*shape=*/data_shape, /*token=*/after_all,
+      /*channel_id=*/channel_id.value_or(0)));
```
Comment (frgossen) [inline xla/service/collective_permute_decomposer.cc:142]: Can you add a todo that we really want to make the channel ID optional in the op? 
Actively working on that

**Review by frgossen — CHANGES_REQUESTED** [review]

File: xla/service/collective_permute_decomposer_test.cc (hunk 1 above)
Comment (shraiysh) [reply xla/service/collective_permute_decomposer_test.cc:290]: Sure, I can do that, but note that I have not edited any of the existing tests themselves. I just wrapped the same code within a lambda and called it with/without channel id. The diff here does not make it obvious. I will send an update soon, but please let me know if you prefer them unchanged. I prefer the filecheck tests too.

File: xla/service/collective_permute_decomposer.cc (hunk 3 above)
Comment (shraiysh) [reply xla/service/collective_permute_decomposer.cc:144]: Done

**Review by frgossen — APPROVED** [review]
Thanks!

**Comment by shraiysh** [comment]
Are there any internal failures with this? @frgossen

**Comment by shraiysh** [comment]
Ping for review! Thanks.

**Comment by frgossen** [comment]
Reed is assigned to this one internally. I think he will get to it after Thanksgiving.

**Comment by shraiysh** [comment]
Closing this, as the change has been merged as a part of a different patch.

**Pull request closed without merging**